src/payment/strategies/googlepay/googlepay-payment-strategy.ts

```diff
@@ -0,0 +1,181 @@
+            throw new InvalidArgumentError('Unable to initialize payment because "options.googlepay" argument is not provided.');
+        }
+
+        this._googlePayOptions = options.googlepay;
+
+        const walletButton = options.googlepay.walletButton && document.getElementById(options.googlepay.walletButton);
+
+        if (walletButton) {
+            this._walletButton = walletButton;
+            this._walletButton.addEventListener('click', this._handleWalletButtonClick);
+        }
+
+        return this._googlePayPaymentProcessor.initialize(this._methodId)
+            .then(() => super.initialize(options));
+    }
+
+    deinitialize(options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (this._walletButton) {
+            this._walletButton.removeEventListener('click', this._handleWalletButtonClick);
+        }
+
+        this._walletButton = undefined;
+
+        return Promise.all([
+            this._googlePayInitializer.teardown(),
+            this._googlePayPaymentProcessor.deinitialize(),
+        ]).then(() => super.deinitialize(options));
+    }
+
+    execute(payload: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        return this._createOrder(this._getPayment(), payload.useStoreCredit, options);
+    }
+
+    private _createOrder(payment: Payment, useStoreCredit?: boolean, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        return this._store.dispatch(this._orderActionCreator.submitOrder({ useStoreCredit }, options))
+            .then(() => this._store.dispatch(this._paymentActionCreator.submitPayment(payment)));
+    }
+
+    private _setExternalCheckoutData(paymentData: GooglePaymentData): Promise<void> {
+        return this._googlePayPaymentProcessor.parseResponse(paymentData)
+            .then((tokenizePayload: TokenizePayload) => {
+                const {
+                    onError = () => {},
+                    onPaymentSelect = () => {},
+                } = this._googlePayOptions;
+
+                return this._paymentInstrumentSelected(tokenizePayload, paymentData.cardInfo.billingAddress)
+                    .then(() => onPaymentSelect())
+                    .catch(error => onError(error));
+            });
+    }
+
+    private _paymentInstrumentSelected(tokenizePayload: TokenizePayload, billingAddress: GooglePayAddress): Promise<InternalCheckoutSelectors> {
+        if (!this._methodId) {
+            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
+        }
+
+        return this._store.dispatch(this._paymentStrategyActionCreator.widgetInteraction(() => {
+            return this._postForm(tokenizePayload, billingAddress);
+        }, { methodId: this._methodId }), { queueId: 'widgetInteraction' });
+    }
+
+    private _postForm(postPaymentData: TokenizePayload, billingAddress: GooglePayAddress): Promise<InternalCheckoutSelectors> {
```

Comment:
I assume `GooglePayPaymentProcessor` is responsible for processing payment data, i.e.: perform any necessary pre-processing and pass the payment data to GooglePay. That's actually what you're doing here in this method. So my question is, why is this method not moved under the payment processor class?